### DISCUSSION TITLE
Underflow/Overflow

### DISCUSSION THREAD

**Discussion by houshangaziziaccelleran** [srsran/srsRAN_Project#637] [General Help]
Hello
Can anyone help with the following issue:
I am testing srsRAN with n310 in FDD/n3/20MHz/2X2 and when I have:
```
  nof_antennas_dl: 1
  nof_antennas_ul: 1
```
It works (obviously with lower throughput than expected.
When I have:
```
  nof_antennas_dl: 2
  nof_antennas_ul: 2
```
then I get a lot of:
`gnb_1             | Late: 246; Underflow: 40; Overflow: 0;`

**Comment by houshangaziziaccelleran** [comment]
Hello,

Here is a brief summary:

    I am using the latest version of srsRAN with an n310 working in the n3 band.
    When a single UE is connected with the following configuration:
```
        nof_antennas_dl: 1
        nof_antennas_ul: 1
``` 
I achieve a DL/UL throughput of approximately 90/70 Mbps.
    However, when I change the configuration to:
```
        nof_antennas_dl: 2
        nof_antennas_ul: 2
``` 
I encounter numerous instances of underflow/overflow, and the DL/UL throughput significantly drops to around 30/50 Mbps.

Logs are big to be uploaded. Please let me know how can I share them. 

Thanks
Houshang

**Comment by brendan-mcauliffe** [comment]
Hi @houshangaziziaccelleran, 

Can you provide some information on the compute platform you are using? And the connecting to the USRP? Also, it would be helpful if you could show the output of `lscpu`. 

You can read more on over/underflows [here](https://files.ettus.com/manual_archive/release_003_008_004/html/page_general.html).

**Comment by houshangaziziaccelleran** [comment]
Hello

Here is the setup:

- srsRAN, the latest version
- n310, using in n3
- BW = 20 MHz
- Cell is wired to Amarisoft's SDR card.

Here is the test:

- Attach 1 UE
- Run Bi-Dir Data

Here is the observation:

When doing 1X1, meaning we have 
```
        nof_antennas_dl: 1
        nof_antennas_ul: 1
```
Then we get something around 90 mbps in the DL and something around 70 mbps in the UL.
But here is the problem: When we do 2X2 meaning we have
```
        nof_antennas_dl: 2
        nof_antennas_ul: 2
```
Then we get very low DL and very low UL (lower than the DL/UL we get for 1X1).

Please find attached `lscpu` printout. This is not CPU issue I think. I have 64 CPUs in my server (the server srsRAN runs on it) and I am using 32 dedicated for srsRAN.

I need to add that I am using DPDK and I have tested without DPDK as well. The results are the same.

I also need to add that with n78, I don;t have such issue. Menaing with n78, 2X2 gives higher throughput than 1X1 (as expected)

[lscpu.txt](https://github.com/user-attachments/files/15512951/lscpu.txt)

**Comment by pgawlowicz** [comment]
hi @houshangaziziaccelleran, any update on this issue? could you share your configs and gnb logs? how do you connect your usrp? do you use 10Gb Ethernet?
